src/components/activity-completion/completion-page-content.tsx

```diff
@@ -1,50 +1,165 @@
+              answerNum++;
+            }
+          });
+        }
+      });
+    }
+    return ({ numAnswers: answerNum, numQuestions: questionNum });
+  };
+
+  useEffect(() => {
+    watchAllAnswers(answerMetas => {
+      setAnswers(answerMetas);
+    });
+  },[]);
+
+  if(!answers) {
+    return (
+      <div className="completion-page-content" data-cy="completion-page-content">
+        <div className="progress-container" data-cy="progress-container">
+          <div className="progress-text">
+            Fetching your data ...
+          </div>
+        </div>
+      </div>  
+    );
+  }
+
+  const progress = activityProgress(activity);
+  const isActivityComplete = progress.numAnswers === progress.numQuestions;
+  const activityTitle = (activityName !== "") || (activityName == null) ? activityName : "the activity";
+  const completionText = activityName ? `${activityName} activity complete!` : "Activity complete!";
+  const activityNum = activityIndex ? activityIndex : 0;
+  const completedActivityProgressText =
+    `Congratulations! You have reached the end of ${activityTitle} and your work has been saved.`;
+  const incompleteActivityProgressText =
+    `It looks like you haven't quite finished ${activityTitle} yet. The answers you've given have been saved.`;
+  const isLastActivityInSequence = activityIndex ? sequence?.activities.length === activityIndex + 1 : false;
+  const nextActivityTitle = !isLastActivityInSequence && sequence?.activities[activityNum + 1].name;
+  const nextActivityDescription = !isLastActivityInSequence && renderHTML(sequence?.activities[activityNum + 1].description || "");
+  const nextStepMainContentTitle = sequence ? (sequence.display_title !== "" ? sequence.display_title : "the sequence") : activityTitle;
+  const completedMainContentNextStepText = `You have completed ${nextStepMainContentTitle} and you may exit now.`;
+  const incompletedMainContentNextStepText =
+    `You haven't completed ${nextStepMainContentTitle} yet. You can go back to complete it, or you can exit.`;
+  let progressText = "";
+  let nextStepText = "";
+
+  if (sequence) {
+    const sequenceComplete = sequenceProgress();
+    
+    if (isLastActivityInSequence) {
+      progressText = isActivityComplete
+        ? completedActivityProgressText + `You have completed all your work for this module!` : incompleteActivityProgressText;
+      nextStepText = sequenceComplete ? completedMainContentNextStepText : incompletedMainContentNextStepText;
+    } else { //if !isLastActivity
+      progressText = isActivityComplete ? completedActivityProgressText : incompleteActivityProgressText;
+
+      nextStepText = "";
+    }
+  } else { //if (!sequence) Assumes is single activity
+    progressText = isActivityComplete ? completedActivityProgressText : incompleteActivityProgressText;
+    nextStepText = isActivityComplete ? completedMainContentNextStepText : incompletedMainContentNextStepText;
+
```

Comment:
probably don't need this line break.  There are a couple more above that I would remove too.